src/global/core/components/animations/animation-utils.ts

```diff
@@ -0,0 +1,173 @@
+
+          keyframe[newKey] = value;
+          delete keyframe[key];
+        } else {
+          const newKey = convertCamelCaseToHypen(key);
+
+          if (newKey !== key) {
+            keyframe[newKey] = value;
+            delete keyframe[key];
+          }
+        }
+      }
+    }
+  });
+
+  return keyframes;
+};
+
+export const getAnimationPrefix = (el: HTMLElement): string => {
+  if (animationPrefix === undefined) {
+    const supportsUnprefixed = (el.style as any).animationName !== undefined;
+    const supportsWebkitPrefix = (el.style as any).webkitAnimationName !== undefined;
+
+    animationPrefix = (!supportsUnprefixed && supportsWebkitPrefix)
+      ? '-webkit-'
+      : '';
+  }
+
+  return animationPrefix;
+};
+
+export const setStyleProperty = (element: HTMLElement, propertyName: string, value: string | null): void => {
+  const prefix = propertyName.startsWith('animation')
+    ? getAnimationPrefix(element)
+    : '';
+
+  element.style.setProperty(prefix + propertyName, value);
+};
+
+export const removeStyleProperty = (element: HTMLElement, propertyName: string): void => {
+  const prefix = propertyName.startsWith('animation')
+    ? getAnimationPrefix(element)
+    : '';
+
+  element.style.removeProperty(prefix + propertyName);
+};
+
+export const animationEnd = (el: HTMLElement | null, callback: (ev?: TransitionEvent) => void): () => void => {
+  let unRegTrans: (() => void) | undefined;
+  const opts: any = {
+    passive: true
+  };
+
+  const unregister = (): void => {
+    if (unRegTrans) {
+      unRegTrans();
+    }
+  };
+
+  const onTransitionEnd = (ev: Event): void => {
+    if (el === ev.target) {
+      unregister();
+      // eslint-disable-next-line callback-return
+      callback(ev as TransitionEvent);
+    }
+  };
+
+  if (el) {
+    el.addEventListener('webkitAnimationEnd', onTransitionEnd, opts);
+    el.addEventListener('animationend', onTransitionEnd, opts);
+
+    unRegTrans = (): void => {
+      el.removeEventListener('webkitAnimationEnd', onTransitionEnd, opts);
+      el.removeEventListener('animationend', onTransitionEnd, opts);
+    };
+  }
+
+  return unregister;
+};
+
+export const generateKeyframeRules = (keyframes: any[] = []): string => keyframes
+  .map((keyframe) => {
+    const {
+      offset
+    } = keyframe;
+
+    const frameString = [];
+
+    for (const property in keyframe) {
+      // eslint-disable-next-line no-prototype-builtins
+      if (keyframe.hasOwnProperty(property) && property !== 'offset') {
+        frameString.push(`${property}: ${keyframe[property]};`);
+      }
+    }
+
+    return `${offset * 100}% { ${frameString.join(' ')} }`;
+  })
+  .join(' ');
+
+const keyframeIds: string[] = [];
+
+export const generateKeyframeName = (keyframeRules: string): string => {
+  let index = keyframeIds.indexOf(keyframeRules);
+
+  if (index < 0) {
+    index = (keyframeIds.push(keyframeRules) - 1);
+  }
+
+  return `ion-animation-${index}`;
```

Comment:
lyne-animation or sbb-animation?

Reply:
Typo from ionic :( will fix here and in a couple of other points..